doc/subport/sonic-sub-port-intf-hld.md

```diff
@@ -96,63 +102,98 @@ A sub port interface shall support the following features:
+    Hence subinterface admin UP is performed only after parent interface is admin UP.
 
 # 2 Schema design
 
 We introduce a new table "VLAN_SUB_INTERFACE" in the CONFIG_DB to host the attributes of a sub port interface.
 For APPL_DB and STATE_DB, we do not introduce new tables for sub port interfaces, but reuse existing tables to host sub port interface keys.
 
 ## 2.1 Configuration
-### 2.1.1 config_db.json
+### 2.1.1 Naming Convention for sub-interfaces:
+
+Since Kernel has netdevice name length restriction to 15, port channel sub-interfaces(Physical interfaces as well in case interface number > 99) cannot follow the same nomenclature as physical sub-interfaces. Hence Long name to  short name conversion needs to be performed for the subinterfaces.
+
+All DB & Kernel netdevice corresponding to the sub-interface should be created with a short name.
+
+All applications interacting/maintaining sub-interfaces should be aware of this mapping to get parent interface properties.
+
+CPP & Python API library will be provided to perform short name to long name conversion and vice versa.
+Please refer to the API library section for details.
+
+Short naming conventions for sub-interfaces will have Ethxxx.yyyy, Poxxx.yyyy format.
+
+All click config CLIs for sub-interfaces will follow long name format.
+
+### 2.1.2 config_db.json
 ```
 "VLAN_SUB_INTERFACE": {
-    "{{ port_name }}.{{ vlan_id }}": {
-        "admin_status" : "{{ adminstatus }}"
+    "{{ port_name }}.{{ subinterface_id }}": {
+        "vlan" : <1-4094>,
+        "admin_status" : "{{ adminstatus }}",
+        "vrf_name" : <vrf-name>
     },
-    "{{ port_name }}.{{ vlan_id }}|{{ ip_prefix }}": {}
+    "{{ port_name }}.{{ subinterface_id }}|{{ ip_prefix }}": {}
 },
 ```
 A key in the VLAN_SUB_INTERFACE table is the name of a sub port, which consists of two sections delimited by a "." (symbol dot).
-The section before the dot is the name of the parent physical port or port channel. The section after the dot is the dot1q encapsulation vlan id.
+The section before the dot is the name of the parent physical port or port channel. The section after the dot is a unique number which uniquely identifies the sub-interface on the parent interface. Sub-interface id value can be in range 1-99999999(Subinterface ID cannot exceed 8 digits).
+
+vlan field identifies the vlan to which the sub-interface is associated using .1Q trunking.
+Note that subinterface_id and vlan_id for a subinterface can be different.
+Unless the user configures vlan_id associated with the subinterface, the subinterface will NOT be provisioned in the SONiC system.
 
-mtu of a sub port interface is inherited from its parent physical port or port channel, and is not configurable in the current design.
+In Click CLI, user will be able to configure the vlan id associated with the sub-interface. If vlan_id is not provided, sub-interface ID will be used as vlan id.
```

Comment:
I think, we should mandate vlan_id configuration even from Click CLI and we should not assume sub-interface id as vlan-id.

Reply:
Updated in next commit where vlan id is mandatory for short name format subinterfaces. But for existing long name format subinterfaces, there is no vlan id hence its not applicable.